specification/dsdl/serializable_types.tex

```diff
@@ -0,0 +1,285 @@
+ordered by bit length is provided below for reference.
+Note that the cast mode specifier is omitted intentionally.
+
+\immediate\write18{rm -f ../latex.tmp}
+\immediate\write18{../render_list_of_void_and_primitive_types.py > ../latex.tmp}
+\immediate\input{../latex.tmp}
+
+\subsection{Array types}
+
+An array type represents an ordered collection of values.
+All values in the collection share the same type, which is referred to as \emph{array element type}.
+The array element type can be any type except:
+\begin{itemize}
+    \item void type;
+    \item array type\footnote{Meaning that nested arrays are not allowed;
+    however, the array element type can be a composite type which in turn may contain arrays.
+    In other words, indirect nesting of arrays is permitted.}.
+\end{itemize}
+
+The number of elements in the array can be specified as a constant at the data type definition time,
+in which case the array is said to be a \emph{fixed-length array}.
+Alternatively, the number of elements can vary between zero and some static limit specified
+at the data type definition time,
+in which case the array is said to be a \emph{variable-length array}.
+Variable-length arrays with unbounded maximum number of elements are not allowed.
+
+Arrays are defined by adding a pair of square brackets after the array element type specification,
+where the brackets contain the \emph{array capacity expression}.
+The array capacity expression must yield a positive integer upon its evaluation;
+any other value or type renders the current DSDL definition invalid.
+
+The array capacity expression can be prefixed with the following character sequences in order to define
+a variable-length array:
+\begin{itemize}
+    \item ``\verb|<|'' (ASCII code 60) --- indicates that the integer value yielded by the array capacity expression
+    specifies the non-inclusive upper boundary of the number of elements.
+
+    \item ``\verb|<=|'' (ASCII code 60 followed by 61) --- same as above, but the upper boundary is inclusive.
+\end{itemize}
+If neither of the above prefixes are provided, the resultant definition is that of a fixed-length array.
+
+\subsection{Composite types}\label{sec:dsdl_composite_types}
+
+\subsubsection{Kinds}
+
+There are two kinds of composite data type definitions: message types and service types.
+All versions of a data type must be of the same kind\footnote{%
+For example, if a data type version 0.1 is of a message kind, all later versions of it must be messages, too.}.
+
+A message data type defines one data schema.
+
+A service data type contains two data schema definitions:
+one for service request object, and one for service response object, in that order.
+The two schemas are separated by the service response marker (``\verb|---|'') on a separate line.
+
+The presence of the service response marker indicates that the data type definition at hand is of the service kind.
+There may be no more than one service response marker in a given definition.
```

Comment:
```suggestion
Only one service response marker shall appear in a given definition.
```

Reply:
Replaced this with "At most one service response marker shall appear in a given definition."